lua/nvim-tree/popup-menu/init.lua

```diff
@@ -0,0 +1,112 @@
+local a = vim.api
+
+local get_node = require('nvim-tree.lib').get_node_at_cursor
+local get_explorer = require('nvim-tree.core').get_explorer
+
+local M = {}
+
+M.is_open = false
+M.win = 0
+M.bufnr = 0
+
+local actions = {
+ ['copy'] = require('nvim-tree.actions.copy-paste').copy 
+}
+
+M.mappings = {
+  ['<Esc>'] = 'close_menu',
+  ['<CR>'] = 'on_selection',
+}
+
+local ns_PopupMenu = a.nvim_create_namespace('ns_NvimTreePopupMenu')
+
+local function create_buf()
+  if vim.fn.bufexists("NvimTreePopupMenuBuf") == 1 then
+    a.nvim_buf_delete(vim.fn.bufnr('NvimTreePopupMenuBuf'), {} )
+  end    
+  local buf = a.nvim_create_buf(false, true)
+  a.nvim_buf_set_name(buf, 'NvimTreePopupMenuBuf')
+  M.bufnr = buf
+  
+  return buf
+end
+
+function M.open_menu()
+  if M.is_open == true then
+    close_menu()
+  end
+
+  create_buf()
+  local win_opts = {
+    width = 10, -- we should set this as NvimExplorer width
+    height = 1, -- TODO: Set this as length of actions, as for now I'm not able to do it.
+    row = 1,
+    col = 0,
+    style = 'minimal',
+    relative = 'cursor'
+  }
+
+  local win_ui = a.nvim_open_win(M.bufnr, true, win_opts)
+  M.is_open = true
+  M.win = win_ui
+  set_actions(actions)
+  
+  -- TODO: Convert this to nvim lua api.
```

Comment:
vim.cmd is fine for autocommands and mappings right now. They are far from stable on 0.7